csi.proto

```diff
@@ -502,6 +510,34 @@ message ControllerServiceCapability {
     RPC rpc = 1;
   }
 }
+
+message ControllerExpandVolumeRequest {
+  // The ID of the volume to expand. This field is REQUIRED
+  string volume_id = 1;
+
+  // the current capacity of the volume in bytes.
+  // This field is OPTIONAL
+  int64 current_capacity_bytes = 2;
+
+  // This field is REQUIRED. This allows CO to specify the
+  // capacity requirements of the volume after expansion
+  CapacityRange capacity_range = 3;
+
+  // Secrets required by the plugin for expanding the volume.
+  // This field is OPTIONAL
+  map<string,string> controller_volume_expand_secrets = 4;
+}
+
+message ControllerExpandVolumeResponse {
+  // Capacity of volume after expansion
+  // REQUIRED
+  int64 capacity_bytes = 2;
+
+  // Whether file system expansion is required for the volume
+  // REQUIRED
+  bool fs_resize_required = 3;
```

Comment:
under what conditions would a CO **not** want an underlying filesystem to be automatically resized? is this field really necessary?

Reply:
For network file systems like - glusterfs, portworx, AzureFile etc there is no file system resize necessary.